src/components/table/daikin-table.ts

```diff
@@ -0,0 +1,462 @@
+ *   </daikin-table-cell>
+ *   <daikin-table-cell slot="cell:name:4">
+ *     Strawberry
+ *     <span slot="subtitle">This is a spring fruit.</span>
+ *   </daikin-table-cell>
+ * </daikin-table>
+ * ```
+ */
+@customElement("daikin-table")
+export class DaikinTable<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends { id: string } = any,
+> extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      --table-color-text: ${unsafeCSS(tableColorText)};
+      --table-color-background: ${unsafeCSS(tableColorBackground)};
+
+      display: block;
+      width: 100%;
+    }
+  `;
+
+  /**
+   * Headers of the table.
+   * - key: The value of `key` corresponds to the key, excluding the id of rows. As a whole array, the value of `key` must be unique. Also, only use alphanumeric characters, `$`, and `_` in the `key`.
+   * - label: This is the text that is displayed in the header cells.
+   * - alignment: The direction in which the characters are aligned can be omitted. If it is omitted, the characters will be aligned to the left.
+   * - leftIcon: In the header cell, you can optionally display an icon to the left of the text.
+   * - sortable: If sortable (`sortable = true`), this specifies whether sorting is performed on this column. If `undefined`, this is considered to be `true`.
+   */
+  @property({ type: Array, attribute: false })
+  headers: HeaderType<Extract<keyof T, string>>[] = [];
+
+  /**
+   * Rows of the table.
+   * An array that uses the element `key` in the column as the key and stores the corresponding value.
+   * As a whole array, the value of `id` should be unique.
+   */
+  @property({ type: Array, attribute: false })
+  rows: T[] = [];
+
+  /**
+   * Whether or not to enable selection of rows.
+   * If `true`, a checkbox will be displayed to the left of each row.
+   */
+  @property({ type: Boolean, reflect: true })
+  selectable: boolean = false;
+
+  /**
+   * Whether or not to enable sorting of the rows.
+   * If `true`, a button for sorting will be displayed to the right of each header cell text.
+   */
+  @property({ type: Boolean, reflect: true })
+  sortable: boolean = false;
+
+  /**
+   * An array of `id`s for the `rows` that have been checked.
+   */
+  @property({ type: Array, attribute: false })
+  selection: string[] = [];
+
+  /**
+   * Sort order of the table.
+   */
+  @property({ type: String, reflect: true })
+  order: "asc" | "desc" | null = null;
+
+  /**
+   * Specify this when you want to customize the sort function.
+   * The function must be ascending, and the return value must be `0`, `1` or `-1`.
+   */
+  @property({ attribute: false })
+  sort:
+    | {
+        [key in keyof T]?: (a: T, b: T, key: key) => number;
+      }
+    | null = null;
+
+  /**
+   * The `key` of the currently sorted column.
+   */
+  @property({ type: String, reflect: true, attribute: "sorted-key" })
+  sortedKey: keyof T | null = null;
```

Comment:
Simplify property name as much as possible.

```suggestion
  @property({ type: String, reflect: true })
  sort: keyof T | null = null;
```